pkg/policy/consumer.go

```diff
@@ -69,81 +111,93 @@ func NewConsumable(id identity.NumericIdentity, lbls *identity.Identity, cache *
 	return consumable
 }
 
+// AddMap adds m to the Consumable's PolicyMaps. This represents
+// the PolicyMap being added for a specific endpoint.
 func (c *Consumable) AddMap(m *policymap.PolicyMap) {
 	c.Mutex.Lock()
 	defer c.Mutex.Unlock()
-	if c.Maps == nil {
-		c.Maps = make(map[int]*policymap.PolicyMap)
+	if c.PolicyMaps == nil {
+		c.PolicyMaps = make(map[int]*policymap.PolicyMap)
 	}
 
 	// Check if map is already associated with this consumable
-	if _, ok := c.Maps[m.Fd]; ok {
+	if _, ok := c.PolicyMaps[m.Fd]; ok {
 		return
 	}
 
 	log.WithFields(logrus.Fields{
 		"policymap":  m,
 		"consumable": c,
 	}).Debug("Adding policy map to consumable")
-	c.Maps[m.Fd] = m
+	c.PolicyMaps[m.Fd] = m
 
 	// Populate the new map with the already established allowed identities from
 	// which ingress traffic is allowed.
 	for ingressIdentity := range c.IngressIdentities {
-		if err := m.AllowIdentity(ingressIdentity.Uint32()); err != nil {
+		if err := m.AllowIdentity(ingressIdentity.Uint32(), Ingress.Uint8()); err != nil {
+			log.WithError(err).Warn("Update of policy map failed")
+		}
+	}
+
+	for egressIdentity := range c.EgressIdentities {
+		if err := m.AllowIdentity(egressIdentity.Uint32(), Egress.Uint8()); err != nil {
 			log.WithError(err).Warn("Update of policy map failed")
 		}
 	}
 }
 
 func (c *Consumable) delete() {
+
 	for ingressIdentity := range c.IngressIdentities {
-		// FIXME: This explicit removal could be removed eventually to
```

Comment:
This comment is being removed, but I think it's still relevant and could probably refer to #2799